packages/main/src/RPA/DocumentAI.py

```diff
@@ -0,0 +1,220 @@
+            lib_kwargs.update(
+                {
+                    "vault_name": name,
+                    "vault_secret_key": key,
+                }
+            )
+            init_kwargs["use_robocorp_vault"] = True
+        elif secret_value:
+            # Vault not used, therefore the provided secret is a file path pointing to
+            #  a service account or token JSON file.
+            if (auth_type or "serviceaccount") == "serviceaccount":
+                secret_type = "service_account"
+            else:
+                secret_type = "token_file"
+            init_kwargs[secret_type] = str(secret_value)
+        if auth_type:
+            lib_kwargs["cloud_auth_type"] = auth_type
+        if region:
+            init_kwargs["region"] = region
+
+        engine = Google(**lib_kwargs)
+        engine.init_document_ai(**init_kwargs)
+        self._engines[EngineName.GOOGLE] = engine
+
+    @staticmethod
+    def _secret_value_to_params(secret_value: str) -> Tuple[str]:
+        return tuple(part.strip() for part in secret_value.split(","))
+
+    def _init_base64(self, secret_value: SecretType):
+        from RPA.Base64AI import Base64AI
+
+        engine = Base64AI()
+        engine.set_authorization(*self._secret_value_to_params(secret_value))
+        self._engines[EngineName.BASE64] = engine
+
+    def _init_nanonets(self, secret_value: SecretType):
+        from RPA.Nanonets import Nanonets
+
+        engine = Nanonets()
+        engine.set_authorization(*self._secret_value_to_params(secret_value))
+        self._engines[EngineName.NANONETS] = engine
+
+    @keyword
+    def switch_engine(self, name: Union[EngineName, str]):
+        name: EngineName = (
+            name if isinstance(name, EngineName) else EngineName(name.lower())
+        )
+        if name not in self._engines:
+            raise RuntimeError(
+                f"can't switch to {name.value!r} engine, please run"
+                f" `Init Engine    {name.value}    ...` first"
+            )
+
+        self._active_engine = name
+
+    @keyword
+    def init_engine(
+        self,
+        name: Union[EngineName, str],
+        secret: Optional[str] = None,
+        vault: Optional[Dict] = None,
+        **kwargs,
+    ):
+        """<summary>
+
+        <params>
+        <example>
+        """
+        if secret and vault:
+            raise ValueError("choose between `secret` and `vault`")
+        elif not (secret or vault):
+            logging.warning("No `secret` or `vault` provided, relying on env vars.")
+
+        init_map = {
+            # Google library needs to be Vault aware too due to its internal way of
+            #  handling secrets from there.
+            EngineName.GOOGLE: functools.partial(self._init_google, vault=vault),
+            # Rest of the engines have a normalized way of picking up secrets either
+            #  provided directly (as string or file path) either through Vault (
+            #  dictionary of parameters).
```

Comment:
Reword this as there's a comma separated arbitrary number of values passed to the authorization keyword now.